libopae/src/mmio.c

```diff
@@ -201,6 +201,10 @@ static fpga_result find_or_map_wm(fpga_handle handle, uint32_t mmio_num,
 			return result;
 		}
 		wm = wsid_find_by_index(_handle->mmio_root, mmio_num);
+		if (!wm) {
+			FPGA_ERR("unable to map wsid for mmio region %d", mmio_num);
+			return FPGA_NO_MEMORY;
```

Comment:
Good fix. Would it make sense to return FPGA_EXCEPTION or FPGA_NOT_FOUND?

Reply:
I think NO_MEMORY is better since this is effectively a memory allocation.